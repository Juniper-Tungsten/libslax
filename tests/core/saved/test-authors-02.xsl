<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax="http://xml.libslax.org/slax" version="1.0" extension-element-prefixes="slax">
  <xsl:variable name="slax-errors" mvarname="errors"/>
  <xsl:variable name="errors" mutable="yes" svarname="slax-errors"/>
  <xsl:template match="/">
    <top>
      <xsl:apply-templates/>
      <xsl:if test="$errors">
        <errors>
          <xsl:copy-of select="$errors"/>
        </errors>
      </xsl:if>
    </top>
  </xsl:template>
  <xsl:template match="text()"/>
  <xsl:template match="author">
    <xsl:variable name="age" select="life-span/died - life-span/born"/>
    <xsl:variable name="slax-problem" mvarname="problem"/>
    <xsl:variable name="problem" mutable="yes" svarname="slax-problem"/>
    <xsl:variable name="slax-tag" mvarname="tag"/>
    <xsl:variable name="tag" select="concat(&quot; (&quot;, $age, &quot; years old)&quot;)" mutable="yes" svarname="slax-tag"/>
    <xsl:choose>
      <xsl:when test="$age &lt; 0">
        <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="problem" svarname="slax-problem" select="&quot;died before born&quot;"/>
      </xsl:when>
      <xsl:when test="$age &lt;= 40">
        <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="problem" svarname="slax-problem" select="&quot;died too young&quot;"/>
      </xsl:when>
    </xsl:choose>
    <xsl:if test="life-span/died - life-span/born &gt; 65">
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="problem" svarname="slax-problem" select="&quot;lived too long&quot;"/>
    </xsl:if>
    <xsl:if test="not($problem) and life-span/born &lt; 1900">
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="problem" svarname="slax-problem" select="&quot;ancient history&quot;"/>
      <slax:set-variable xmlns:slax="http://xml.libslax.org/slax" name="tag" svarname="slax-tag" select="concat(&quot; (c. &quot;, life-span/born, &quot;)&quot;)"/>
    </xsl:if>
    <xsl:if test="$problem">
      <slax:append-to-variable xmlns:slax="http://xml.libslax.org/slax" name="errors" svarname="slax-errors">
        <error>
          <name>
            <xsl:value-of select="concat(name/first, &quot; &quot;, name/last)"/>
          </name>
          <message>
            <xsl:value-of select="concat($problem, $tag)"/>
          </message>
        </error>
      </slax:append-to-variable>
    </xsl:if>
  </xsl:template>
</xsl:stylesheet>
