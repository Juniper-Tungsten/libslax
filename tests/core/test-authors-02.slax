version 1.1;

mvar $errors;

match / {
    <top> {
	apply-templates;

	if ($errors) {
	    <errors> { copy-of $errors; }
	}
    }
}

match text() { }

match author {
    var $age = life-span/died - life-span/born;
    mvar $problem;
    mvar $tag = " (" _ $age _ " years old)";

    if ($age < 0) {
	set $problem = "died before born";
    } else if ($age <= 40) {
	set $problem = "died too young";
    }

    if (life-span/died - life-span/born > 65) {
	set $problem = "lived too long";
    }

    if (not($problem) && life-span/born < 1900) {
	set $problem = "ancient history";
	set $tag = " (c. " _ life-span/born _ ")";
    }

    if ($problem) {
	append $errors += <error> {
	    <name> name/first _ " " _ name/last;
	    <message> $problem _ $tag;
	}
    }
}
