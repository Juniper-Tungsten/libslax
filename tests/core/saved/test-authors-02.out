<?xml version="1.0"?>
<top>
  <errors>
    <error>
      <name>Frank Arnau</name>
      <message>lived too long (82 years old)</message>
    </error>
    <error>
      <name>E.C. Bentley</name>
      <message>lived too long (81 years old)</message>
    </error>
    <error>
      <name>Earl Derr Biggers</name>
      <message>ancient history (c. 1884)</message>
    </error>
    <error>
      <name>Collin Brooks</name>
      <message>lived too long (66 years old)</message>
    </error>
    <error>
      <name>John Dickson Carr</name>
      <message>lived too long (72 years old)</message>
    </error>
    <error>
      <name>Raymond Chandler</name>
      <message>lived too long (71 years old)</message>
    </error>
    <error>
      <name>Leslie Charteris</name>
      <message>lived too long (86 years old)</message>
    </error>
    <error>
      <name>G. K. Chesterton</name>
      <message>ancient history (c. 1874)</message>
    </error>
    <error>
      <name>Agatha Christie</name>
      <message>lived too long (86 years old)</message>
    </error>
    <error>
      <name>Brian Cleeve</name>
      <message>lived too long (82 years old)</message>
    </error>
    <error>
      <name>Michael Collins</name>
      <message>lived too long (81 years old)</message>
    </error>
    <error>
      <name>Amanda Cross</name>
      <message>lived too long (77 years old)</message>
    </error>
    <error>
      <name>Arthur Conan Doyle</name>
      <message>lived too long (71 years old)</message>
    </error>
    <error>
      <name>Rampo Edogawa</name>
      <message>lived too long (71 years old)</message>
    </error>
    <error>
      <name>Erle Stanley Gardner</name>
      <message>lived too long (81 years old)</message>
    </error>
    <error>
      <name>Chester Gould</name>
      <message>lived too long (85 years old)</message>
    </error>
    <error>
      <name>Dashiell Hammett</name>
      <message>lived too long (67 years old)</message>
    </error>
    <error>
      <name>Edmund Wallace Hildick</name>
      <message>lived too long (76 years old)</message>
    </error>
    <error>
      <name>Michael Innes</name>
      <message>lived too long (88 years old)</message>
    </error>
    <error>
      <name>Dennis Lynds</name>
      <message>lived too long (81 years old)</message>
    </error>
    <error>
      <name>John D. MacDonald</name>
      <message>lived too long (70 years old)</message>
    </error>
    <error>
      <name>Ross MacDonald</name>
      <message>lived too long (68 years old)</message>
    </error>
    <error>
      <name>Ngaio Marsh</name>
      <message>lived too long (87 years old)</message>
    </error>
    <error>
      <name>Gladys Mitchell</name>
      <message>lived too long (82 years old)</message>
    </error>
    <error>
      <name>Ellis Peters</name>
      <message>lived too long (82 years old)</message>
    </error>
    <error>
      <name>Edgar Allan Poe</name>
      <message>died too young (40 years old)</message>
    </error>
    <error>
      <name>Dorothy L. Sayers</name>
      <message>ancient history (c. 1893)</message>
    </error>
    <error>
      <name>Georges Simenon</name>
      <message>lived too long (86 years old)</message>
    </error>
    <error>
      <name>Rex Stout</name>
      <message>lived too long (89 years old)</message>
    </error>
    <error>
      <name>Josephine Tey</name>
      <message>ancient history (c. 1896)</message>
    </error>
    <error>
      <name>H. Russell Wakefield</name>
      <message>lived too long (76 years old)</message>
    </error>
    <error>
      <name>Patricia Wentworth</name>
      <message>lived too long (83 years old)</message>
    </error>
  </errors>
</top>
